lib/flag_shih_tzu.rb

```diff
@@ -335,7 +335,7 @@ def check_flag_column(colmn, custom_table_name = table_name)
       has_ar = (!!defined?(ActiveRecord) && respond_to?(:descends_from_active_record?))
       # Supposedly Rails 2.3 takes care of this, but this precaution
       #   is needed for backwards compatibility
-      has_table = has_ar ? connection.tables.include?(custom_table_name) : true
+      has_table = has_ar ? connection.data_sources.include?(custom_table_name) : true
```

Comment:
I assume this would break compatibility with older versions of Rails.  I need to decide if I want to break compatibility and create a legacy branch for those older versions.